src/System.Private.Xml/tests/XmlReader/Tests/ReaderEncodingTests.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using Xunit;
+
+namespace System.Xml.Tests
+{
+    public class ReaderEncodingTests
+    {
+        [Fact]
+        public static void ReadWithSurrogateCharAndInvalidChar()
+        {
+            var bytes = new byte[] { 60, 0, 0, 0, 0, 34, 1, 0, 62, 100, 60, 47, 97, 62, 10 };
```

Comment:
if there is some reasonable textual representation for these I would prefer if you used that. I.e.:
`Encoding.Unicode.GetBytes("<?xml\u0123")`

Reply:
Most of them doesn't have reasonable textual representation, actually the error happening while trying to convert these bytes to character

Reply:
added comment about which char is surrogate, which is invalid, which is normal char